packages/server/src/threads/query.ts

```diff
@@ -136,21 +136,23 @@ class QueryRunner {
       pagination = output.pagination
     }
 
-    // transform as required
-    if (transformer) {
+    // We avoid invoking the transformer if it's trivial because there is a cost
+    // to passing data in and out of the isolate, especially for MongoDB where
+    // we have to bson serialise/deserialise the data.
+    const hasTransformer =
+      transformer != null &&
+      transformer.length > 0 &&
+      transformer.trim() !== "return data" &&
+      transformer.trim() !== "return data;"
+
+    if (transformer && hasTransformer) {
       transformer = iifeWrapper(transformer)
       let vm = new IsolatedVM()
       if (datasource.source === SourceName.MONGODB) {
         vm = vm.withParsingBson(rows)
```

Comment:
Should `hasTransformer` only skip this? Can we run the `return data` without the isolated-vm? 🤔

Reply:
I'm pretty sure if the transformer just returns the data as-is, it can be skipped. If we only skipped `vm.withParsingBson` we'd have error about trying to pass objects across the isolate boundary.